core/src/main/java/com/distkv/core/concepts/DistkvConcepts.java

```diff
@@ -1,35 +1,38 @@
 package com.distkv.core.concepts;
 
-import com.distkv.common.exception.DistkvKeyDuplicatedException;
 import com.distkv.common.exception.KeyNotFoundException;
 import com.distkv.common.utils.Status;
 import com.distkv.core.DistkvMapInterface;
 
 public abstract class DistkvConcepts<T> {
 
   // The Reference of the key value map.
-  protected DistkvMapInterface<String, Object> distkvKeyValueMap;
+  protected DistkvMapInterface<String, DistkvValue> distkvKeyValueMap;
 
-  protected DistkvConcepts(DistkvMapInterface<String, Object> distkvKeyValueMap) {
+  protected DistkvConcepts(DistkvMapInterface<String, DistkvValue> distkvKeyValueMap) {
     this.distkvKeyValueMap = distkvKeyValueMap;
   }
 
-  public void put(String key, T value) {
-    if (distkvKeyValueMap.containsKey(key)) {
-      throw new DistkvKeyDuplicatedException(key);
-    }
-    distkvKeyValueMap.put(key, value);
-  }
+  public abstract void put(String key, T value);
 
+  /**
+   * According to the key to return the corresponding value information.
+   * @param key The key.
+   * @return Value corresponding to the data type.
+   */
   @SuppressWarnings("unchecked")
   public T get(String key) {
     if (!distkvKeyValueMap.containsKey(key)) {
       throw new KeyNotFoundException(key);
     }
-
-    return (T) distkvKeyValueMap.get(key);
+    return (T)as(distkvKeyValueMap.get(key));
```

Comment:
Why we should cast here?

Reply:
Because `distkvKeyValueMap.get(key)` returns not a specific T, but an Object object